### DISCUSSION TITLE
How to find a Decomposition of unstable compound into stable compounds (Pymatgen, materials project)

### DISCUSSION THREAD

**Discussion by sachin-rangaswamy** [materialsproject/pymatgen#3599] [Q&A] [answered]
Can anybody help with this!! I want know the module in Pymatgen where it can find the decomposition of unstable compound(random) into stable compounds like how the materials project does. I have attached the picture of reference where LiScVO₄
mp-754728 being unstable compound decomposes into stable compounds 5⁄42[V₂O₃]+ 8⁄21[Li₃VO₄] + 1⁄7[VO₂] + 5⁄14[Sc₂O₃] . Please if someone know the method or the module in Pymatergen to how to access this, it would be a great help, thank you! . <img width="242" alt="sC" src="https://github.com/materialsproject/pymatgen/assets/121004207/5053ba64-33d9-4f13-b9e4-7444b2f28feb">

**Comment by janosh** [comment, marked as answer]
this data comes from `PhaseDiagram.get_decomp_and_phase_separation_energy`:

https://github.com/materialsproject/pymatgen/blob/dca98bec837e0186fb3aae4d3e6880a39013945d/pymatgen/analysis/phase_diagram.py#L831-L855

**Reply by sachin-rangaswamy** [reply]
Thank you very much!!

**Reply by sachin-rangaswamy** [reply]
In order to run this function I understand that I need two arguments and I am able to produce self argument. But how do I produce PDEntry for the entry argument ?

**Reply by janosh** [reply]
[`PDEntry`](https://github.com/materialsproject/pymatgen/blob/91b6a6d74a98e1a60358961c07929efe4547ac7b/pymatgen/analysis/phase_diagram.py#L50) just requires a composition and an energy. those can come from anywhere. in the case of MP it's usually the PBE or r2SCAN DFT energy
